File: dsa/stack_and_queue/1544_Make_The_String_Great.java
package stack_and_queue;
import java.util.*;

class Solution {
    public String makeGood(String s) {
        Stack<Character> charStack = new Stack<Character>();
        StringBuilder result = new StringBuilder();
        for (int i = 0; i < s.length(); i++){
            if (charStack.isEmpty() == false 
                && Character.toUpperCase(s.charAt(i)) == Character.toUpperCase(charStack.peek())
                && (int) s.charAt(i) != (int) charStack.peek())   {
                    charStack.pop();
                }
            else{
                charStack.push(s.charAt(i));
            }
        }

        while (charStack.isEmpty()== false){
            result.append(charStack.pop());
        }
        return result.reverse().toString();
    }
}

File: dsa/9.stack_and_queue/1544_Make_The_String_Great.py
import typing as t

class Solution:
    def makeGood(self, s: str) -> str:
        stack: t.List[str] = []
        for char in s:
            if stack and char != stack[-1] and char.upper() == stack[-1].upper():
                stack.pop()
            else:
                stack.append(char)
        return "".join(stack)